lib/domain/service/file_generator_service/screen_generators/stateful_screen_generator.dart

```diff
@@ -0,0 +1,92 @@
+import 'dart:io';
+
+import 'package:onix_flutter_bricks/domain/service/base/base_generation_service.dart';
+import 'package:onix_flutter_bricks/domain/service/base/params/base_generation_params.dart';
+import 'package:onix_flutter_bricks/domain/service/file_generator_service/screen_generators/gen/stateful_screen_code_content.dart';
+import 'package:onix_flutter_bricks/domain/service/file_generator_service/screen_generators/params/screen_generator_params.dart';
+import 'package:onix_flutter_bricks/util/enum/project_router.dart';
+import 'package:recase/recase.dart';
+
+class StatefulScreenGenerator extends ScreenGenerationService {
+  final _screenCodeContent = StatefulScreenCodeContent();
+
+  @override
+  Future<bool> generate(BaseGenerationParams params) async {
+    if (params is! ScreenGeneratorParams) {
+      return false;
+    }
+
+    String screenName = params.screen.name.snakeCase;
+
+    if (screenName.endsWith('_screen')) {
+      screenName = screenName.substring(0, screenName.length - 7);
```

Comment:
@DemeoStep, this appears so often in all ScreenGenerationService classes that it makes sense to put it in a separate superclass method or extension